rust-runtime/aws-smithy-http/src/operation/error.rs

```diff
@@ -185,3 +186,9 @@ impl Error for BuildError {
         }
     }
 }
+
+impl From<BuildError> for DeserializeError {
+    fn from(build_error: BuildError) -> Self {
+        DeserializeError::custom_source("deserialization failed due to a build error", build_error)
```

Comment:
should we make a nicer error here? e.g `because a required field was missing`, or something?

Reply:
`builderror` is just a weird name to expose to a customer